### ISSUE TITLE
Unlimited storage pickling

### ISSUE THREAD

**Issue by henryiii** [scikit-hep/boost-histogram#138]
The following test fails (disabled for now):

```python
s = bh.storage.unlimited()
pickle.loads(pickle.dumps(s, -1))
```

(And, the unlimited storage in a histogram fails, too, of course)

The error is:

```
RuntimeError: make_tuple(): unable to convert argument of type 'boost::histogram::unlimited_storage<std::__1::allocator<char> >::buffer_type' to Python object
```

So the parts in storage do not have Python types, causing it to fail pickling.

**Comment by HDembinski** [comment]
I will fix this along with a major update on the serialization code. The serialization code in boost::histogram is now independent of Boost.Serialization in the develop branch and can now be reused here.

**Comment by HDembinski** [comment]
Where is the disabled test? I want to check whether this is fixed now (it should).

**Comment by henryiii** [comment]
https://github.com/scikit-hep/boost-histogram/blob/b74967f76430a94a4ee1a2baa5af124795a78117/tests/test_pickle.py#L32

**Comment by HDembinski** [comment]
fixed by #166
